server/src/main/java/org/opensearch/index/mapper/MapperService.java

```diff
@@ -228,7 +229,7 @@ public MapperService(
         this.idFieldDataEnabled = idFieldDataEnabled;
 
         if (INDEX_MAPPER_DYNAMIC_SETTING.exists(indexSettings.getSettings())) {
-            throw new IllegalArgumentException("Setting " + INDEX_MAPPER_DYNAMIC_SETTING.getKey() + " was removed after version 6.0.0");
+            logger.warn("Setting {} was removed after version 6.0.0, its value will be ignored.", INDEX_MAPPER_DYNAMIC_SETTING.getKey());
```

Comment:
@peternied I am not sure I understand this change: we do support migration from ES 7.10.x minimum, how come we run into the indices that are still having this settings after migration to 7.10?

Reply:
I suspect that there are still paths that allow modification of index settings that aren't creating an instance of the MapperService allowing bypassing of this check, maybe snapshot/restore?  When I encountered this issue it was of an existing OS 2.5 cluster attempting to upgrade to OS 2.9 - 🤷 

Any ideas that come to mind for you?

Reply:
> Any ideas that come to mind for you?

Thank Peter, no ideas, really, but the mechanism the unsupported settings are going through in OpenSearch is archiving, so I am really curious to under how it happened before trying to workaround it.

Reply:
How would you feel about me opening an issue for the follow up investigation and potential mitigation?

Reply:
> How would you feel about me opening an issue for the follow up investigation and potential mitigation?

:+1: I thanks @peternied , I would suggest to hold on merging this change till we understand what is happening, there are high chances to make things worst

Reply:
From high level, I am not sure if ignoring the exception is the right approach as this may lead to inconsistencies in the index later. As @reta said, we need to dig deeper and get to the cause of the exception first.

Reply:
I'll need to do a further investigation and attempt to reproduce - which might take a little while.  I've marked this PR as draft and when I've got a satisfying answer/resolution to this thread I'll un-draft.

Reply:
@reta @vikasvb90 I was able to find a way to get an index into this state, filed an issue for the root cause [1].  I'm not sure how we should attempt to solve that problem - but in the case of this setting ignoring the value seems safe.  

However, there are other problems such as when two settings values need to be in agreement with one another its unclear what the outcome should be.  I'd prefer to address those on a case by case basis outside the scope of this change.  What do you think?

- [1] https://github.com/opensearch-project/OpenSearch/issues/12350

Reply:
@peternied thanks a lot, @peternied , this is by all means an outstanding investigation. I am not sure that ignoring the value is the way to go BUT we should certainly handle that, there are few options I have in mind:
 - we do have support for settings archival (but not setting values archival which is what we have here), we may replace the "archived" (unsupported) value with the respective substitution
 - also, we may resurrect this setting value to support older indices (as per our compatibility policy) but disallow its usages for new indices, that should probably be done this way from the beginning

Reply:
It is a bug that we 'break' an index so it cannot be actioned on if it has this setting.  Switching the exception to a deprecation message is an improvement that helps unblock customers migrating from very old versions to modern versions of OpenSearch - what this pull request does.

@reta I'm going to push back against handling this scenario any more gracefully - OpenSearch's support policy is between major versions such as 1.x onto 2.x this case is far outside that window.  You seem passionate about this legacy support case - what do you think of capturing that as a new issue that is handled outside of this PR?

Reply:
In 2019 in ES6 the feature deprecation was announced. In ES7 the hard block via exception was added, predating OpenSearch.

Reply:
Oh I see, Thanks a lot @peternied , we somehow missed that. 

> Switching the exception to a deprecation message is an improvement that helps unblock customers migrating from very old versions to modern versions of OpenSearch - what this pull request does.

I think this is the best we can do, I agree with you. Could we use `DeprecationLogger` for that (`deprecationLogger`)?